src/ui.rs

```diff
@@ -1,23 +1,21 @@
-use std::hash::Hash;
+use std::{collections::HashMap, hash::Hash};
 
 use egui::{
-    ahash::{HashMap, HashMapExt},
-    collapsing_header::paint_default_icon,
-    emath::Rot2,
-    epaint::Shadow,
-    pos2, vec2, Align, Color32, Frame, Layout, Modifiers, PointerButton, Pos2, Rect, Sense, Shape,
-    Stroke, Style, Ui, Vec2,
+    collapsing_header::paint_default_icon, epaint::Shadow, pos2, vec2, Align, Color32, Frame, Id,
+    Layout, Modifiers, PointerButton, Pos2, Rect, Sense, Shape, Stroke, Style, Ui, Vec2,
 };
 
 use crate::{InPin, InPinId, Node, OutPin, OutPinId, Snarl};
 
+mod background_pattern;
 mod pin;
 mod state;
 mod viewer;
 mod wire;
 mod zoom;
 
 use self::{
+    background_pattern::BackgroundPattern,
```

Comment:
It should be re-exported

Reply:
Not sure what you mean

Reply:
It is not visible from other crates.
You should put it into `pub use self::{}` block